src/unleash_client_manager.ts

```diff
@@ -9,13 +9,58 @@ import { registerStrategies } from "./strategy/appStudioStrategies";
 // map value = Unleash client
 const unleashClientMap = new Map<string, Unleash>();
 
+export interface InitializeAttempt {
+  numAttempt: number;
+  timeAttempt: number;
+  isBlocked: boolean;
+}
+
+// map key = extensionName
+// map value = number of attempts to establish unleash client
+const initializeAttemptMap = new Map<string, InitializeAttempt>();
+const limitNumber = 2;
+
+function handleUnauthorisedError(extensionName: string): void {
+  let num = initializeAttemptMap.get(extensionName)?.numAttempt;
+  num = (num || 0) + 1;
+  if (num < limitNumber) {
+    initializeAttemptMap.set(extensionName, { numAttempt: num, timeAttempt: 0, isBlocked: false });
+  }
+  if (num == limitNumber) {
+    initializeAttemptMap.set(extensionName, { numAttempt: num, timeAttempt: Date.now(), isBlocked: true });
+  }
+}
+
+// handle unauthorised calls. Block calls for client initialization while in block time period.
+export function handleUnauthorisedCalls(extensionName: string, initializeAttemptMap: Map<string, InitializeAttempt>): void {
+  const initializationInfo = initializeAttemptMap.get(extensionName);
+  // check that client initialization will be not called while in block period
```

Comment:
Please remove comments. Code should speak for himself.

Reply:
done